Dockerfile.base

```diff
@@ -0,0 +1,60 @@
+ARG BASE_IMAGE=debian:latest
```

Comment:
I would prefer that we put `Dockerfile` into a folder hierarchy over giving them some convention-based file extension. For example:

```
build/
  base/
    Dockerfile
```

My rationale is that developer tooling recognizes `Dockerfile` but you get a mixed experience with most other conventions (as evidenced by the lack of syntax highlighting in this pull request).

Reply:
ah I think `base.Dockerfile` might have been picked up correctly.

but yes that's a good point. I think as this image is the most universal probably the most sensible would be to move `/Dockerfile.base` -> `/Dockerfile` and `/Dockerfile` -> `/rust/Dockerfile`

Reply:
Also: if there is a Dockerfile out of context on its own in a folder somewhere, do add a `README.md` to explain its meaning!

Reply:
- sure thing https://github.com/commontoolsinc/system/issues/65

what are your thoughts on:
- `/Dockerfile.base` -> `/Dockerfile`
- `/Dockerfile` -> `/rust/Dockerfile`

Reply:
That's fine for now if you add a README per https://github.com/commontoolsinc/system/pull/62#discussion_r1690394769

Reply:
see: https://github.com/commontoolsinc/system/pull/62/commits/bda533249144429d6de53ab454505d92d51ac840